docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+    item: Item,
+    delta_in: RationalTime = None,   #< Duration of change
+    delta_out: RationalTime = None,  #< Duration of change
+)
+````
+- In the example image, `A` is the item and `delta_out` is a negative RationalTime. The edit command augments `B` without changing it's `source_range().end_time_exclusive()`
+- If `delta_in` is supplied, the item to the left of `A` would be modified (if any) - otherwise we may need to add a `Gap` / fill template.
+
+---
+
+## 3/4 Point Edit
+![3_4 Point Edit](../_static/edit/08_pointedit.png)
+- The most complex - this "fills" a gap based on a source in/out point _or_ track in/out point
+- Often used to patch in items as edit is built
+- Note: This can be accomplished by a conjunction of commands above
+
+### API
+```py
+"""
+Args:
+    item: The item to place onto the track
+    track: Track that will now own this item
+    replace: Item (or RT?) that we will effectively replace
+    reference_point: For 4 point editing, the reference point dictates what
+        transform to use when running the fill.
+
+        Options:
+            - Source   : Don't modify the source, overwrite if required
+            - Sequence : Don't modify the sequence, trim item if required
+            - Fit      : Apply Time Effect
+"""
+otio.algorithms.fill(
+    item: Item,
+    track: Track,
+    replace: Item = None, # Alternatively, we could have a parent-coord
+                          # RationalTime to find the item for us
+    reference_point: otio.algorithms.ReferencePoint.Source
+)
+```
+- This may still require some additional tinkering.
+
+
+# Expanded Implementation
+
+## Proposal
+For editing commands, we should strive to do all validation and assert that everything "works" before committing anything on the timeline. A simple atomic command structure will provide that level of sophistication.
+
+```cpp
+class EditEvent {
+public:
+    EditEventKind kind; // e.g. Insert, Append, Remove, Modify
+    Retainer<Composition> parent;
+    Retainer<Item> composable;
+
+    // ... Additional fields to execute the above as required
+
+    bool run(ErrorStatus* error_status);
+    bool revert();
+};
+```
+An event is atomic in nature and does "one thing" to an `Item`. Each edit maneuver (e.g. `otio.algorithms.overwrite(...)`) would generate a vector of these events that can be played forward and, possibly, backward. The result of them collectively is the commands result.
+
+```cpp
+for (EditEvent &event: events) {
+    event.run(error_status);
+    if (*error_status) {
+        for (auto it = completed_events.rbegin(); /*...*/)
+            (*it).revert();
+        break;
+    }
+    completed_events.push_back(event);
+}
+```
+
+## Overall
+Many of the commands mentioned have common code paths and math that is required. We can streamline many of the placement commands into a single call with different options. We can then expose that as a raw edit platform while giving users the common algorithms for ease-of-use.
```

Comment:
I personally lean away from "uber" functions, and more to concise, does-one-thing functions. Do you have something specific in mind here, or a mock-up example?

Reply:
It's for the case of actions like `overwrite` and `insert` which have the same functionality if placing an item passed the tracks current duration. I'll try to insert additional examples in the next push 👍

Reply:
Okay, I'll look forward to that :)